src/components/editor/diacritics/DiacriticsSelection.jsx

```diff
@@ -0,0 +1,73 @@
+// Copyright 2019 Stanford University see LICENSE for license
+
+import React from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import { closeDiacritics, itemsSelected, setDiacriticsButtons } from 'actions/index'
+import shortid from 'shortid'
+import CharacterButton from './CharacterButton'
+import VocabChoice from './VocabChoice'
+import specialcharacters from '../../../../static/specialcharacters.json'
+
+
+const DiacriticsSelection = () => {
+  const dispatch = useDispatch()
+  const characterButtons = useSelector(state => state.selectorReducer.editor.diacritics.buttons)
+  const show = useSelector(state => state.selectorReducer.editor.diacritics.show)
+  const targetReduxPath = useSelector(state => state.selectorReducer.editor.diacritics.reduxPath)
+
+  let cssClasses
+  if (show) {
+    cssClasses = ['sticky-top', 'bg-white']
+  } else {
+    return null
+  }
+
+  const selectVocabulary = (event) => {
+    const vocabChoice = event.target.value
+    if (vocabChoice === undefined) return null
+    const buttons = []
+    specialcharacters[vocabChoice].characters.map((char) => {
+      buttons.push(<CharacterButton key={shortid.generate()}
+                                    character={char} />)
+    })
+    dispatch(setDiacriticsButtons(buttons))
```

Comment:
As discussed, would prefer not to store this in redux state.

Reply:
Done, reverted to using local component state for character buttons. **Note** this means that we preserve the state of the `DiacriticsSelection` (buttons and selected character set) between invocations by the user.